### PR TITLE
tests: PIT flaky component monitoring test

### PR DESCRIPTION
**Pull request by assafad**
### What this PR does
`VirtControllerRESTErrorsBurst` and `VirtControllerRESTErrorsHigh` alerts e2e test is failing, probably due to leftovers from other test cases. This PR puts the testcase on `PIT`, until we investigate it further, in order to release the monitoring lane.

Fixes https://issues.redhat.com/browse/CNV-38827

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: tests/monitoring/component_monitoring.go
Comment (EdDev) [inline tests/monitoring/component_monitoring.go:238]: Perhaps it will be better to check this after deleting the virt-controller CRB and not after both are deleted.

**Review by EdDev — CHANGES_REQUESTED** [review]
Could you please update the commit message with the reason of this change?

The test touched is destructive, if it flakes on the current tests which have been merged it is obviously going to flake now on another test (that will run after this new merged one).

I have not seen in the test teardown what checks that the error alerts are gone. I would expect the test to wait for them to be gone before proceeding to avoid the next test from starting in a bad setup.

**Comment by assafad** [comment]
> Could you please update the commit message with the reason of this change?
> 
> The test touched is destructive, if it flakes on the current tests which have been merged it is obviously going to flake now on another test (that will run after this new merged one).
> 
> I have not seen in the test teardown what checks that the error alerts are gone. I would expect the test to wait for them to be gone before proceeding to avoid the next test from starting in a bad setup.

@EdDev Seems like this change doesn't help anyway, as the test keeps failing on CI. Locally the original test passes, so I can't say what's the issue for now. I will move the PR to draft until we find the issue.

**Comment by EdDev** [comment]
> @EdDev Seems like this change doesn't help anyway, as the test keeps failing on CI. Locally the original test passes, so I can't say what's the issue for now. I will move the PR to draft until we find the issue.

Most-likely the problem is with tests that run before this one and the cluster does not recover yet. Or the recovery is raceful.
I suggest to remove all destructive tests and provide unit tests alternatives to them.

**Comment by enp0s3** [comment]
@EdDev Hey, thanks for the review. The lane is optional, its being triggered upon monitoring related changes. It was stable in the past, now with 1.29 it constantly fails on `AfterEach`. 

The PR followed my suggestion to squash the test into a single `It` spec, thus reducing the runtime (currently there is overhead with scaling in/out of the operator, and re-creation of the VMI in each spec).

AFAIK we already cover it with unit testing, but the unit testing mocks the metrics, I think it still worth to keep the e2e since we want to make sure the metric values in real life meets the mocked expectations.

**Comment by EdDev** [comment]
> AFAIK we already cover it with unit testing, but the unit testing mocks the metrics, I think it still worth to keep the e2e since we want to make sure the metric values in real life meets the mocked expectations.

Merging e2e tests when possible is great.
The fact these are triggered only when one touches a specific file, is also a nice optimization.

However, I still do not think they worth it at this test level.
But this is just one opinion and it is clear to me it will be hard to convince maintainers to remove e2e tests in general.

**Review by dhiller — APPROVED** [review]
/approve

**Comment by EdDev** [comment]
> `VirtControllerRESTErrorsBurst` and `VirtControllerRESTErrorsHigh` alerts e2e test is failing, probably due to leftovers from other test cases. This PR puts the testcase on `PIT`, until we investigate it further, in order to release the monitoring lane.

It would have been preferred to see this explanation on the commit message.

**Comment by assafad** [comment]
> > `VirtControllerRESTErrorsBurst` and `VirtControllerRESTErrorsHigh` alerts e2e test is failing, probably due to leftovers from other test cases. This PR puts the testcase on `PIT`, until we investigate it further, in order to release the monitoring lane.
> 
> It would have been preferred to see this explanation on the commit message.

Done, @enp0s3 can you please re-approve?

**Pull request merged**
